test/lib/global_uid_test.rb

```diff
@@ -231,34 +301,117 @@ def table_exists?(connection, table)
+            # before the current thread receives its next one. We rely on the gap being divisible by the configured increment
+            test_unique_ids(10)
+            assert_empty(@notifications)
+          end
+        end
+
+        describe "and only one server reports a value other than what's configured" do
+          it "notifies the client when configuration incorrect during initialization" do
+            with_modified_connections(increment: 42, servers: ["test_id_server_1"]) do
+              test_unique_ids(32)
+              assert_includes(@notifications, GlobalUid::InvalidIncrementPreventionException)
+            end
+          end
+
+          it "notifies the client and continues with the other connection" do
+            con = GlobalUid::Base.get_connections.first
+            con.execute("SET SESSION auto_increment_increment = 42")
+
+            # Trigger the exception, one call may not hit the server, there's still a 1/(2^32) chance of failure.
+            test_unique_ids(32)
+            assert_includes(@notifications, GlobalUid::InvalidIncrementException)
+          end
+
+          it "notifies the client and continues when attempting to generate many UIDs" do
+            con = GlobalUid::Base.get_connections.first
+            con.execute("SET SESSION auto_increment_increment = 42")
+
+            # Trigger the exception, one call may not hit the server, there's still a 1/(2^32) chance of failure.
+            32.times { WithGlobalUID.generate_many_uids(10) }
+            assert_includes(@notifications, GlobalUid::InvalidIncrementPreventionException)
+          end
+        end
+      end
     end
 
     describe "With a timing out server" do
-      before do
-        reset_connections!
-        @a_decent_cx = GlobalUid::Base.new_connection(GlobalUid::Base.global_uid_servers.first, 50)
-        ActiveRecord::Base.stubs(:mysql2_connection).raises(GlobalUid::ConnectionTimeoutException).then.returns(@a_decent_cx)
-        @connections = GlobalUid::Base.get_connections
+      def with_timed_out_connection(server:, duration:)
+        modified_connection = Proc.new do |config|
+          if config["database"].include?(server)
+            raise GlobalUid::ConnectionTimeoutException if duration > Time.now
+          end
+
+          ActiveRecord::Base.__minitest_stub__mysql2_connection(config)
+        end
+        ActiveRecord::Base.stub :mysql2_connection, modified_connection do
+          reset_connections!
+          yield
+        end
       end
 
       it "limp along with one functioning server" do
-        assert_includes @connections, @a_decent_cx
-        assert_equal GlobalUid::Base.global_uid_servers.size - 1,  @connections.size, "get_connections size"
+        with_timed_out_connection(server: "test_id_server_1", duration: Time.now + 10.minutes) do
+          test_unique_ids(10)
+          assert_equal 1, GlobalUid::Base.get_connections.size
```

Comment:
Can we also make an assertion on the connection’s name? Or is that too vague and obvious to test?